src/app/components/account-dialog/account-dialog.component.ts

```diff
@@ -0,0 +1,268 @@
+            .pipe(
+                filter(([asset, servicesToken]) => asset !== 'ETH'),
+                switchMap(([selectedToken, servicesToken]) => {
+                    if (selectedToken.address === servicesToken.address) {
+                        return this.userDepositService.servicesToken$;
+                    }
+                    return this.tokenPollingService.getTokenUpdates(
+                        selectedToken.address
+                    );
+                }),
+                map((token) => token.balance),
+                takeUntil(this.ngUnsubscribe)
+            )
+            .subscribe((balance) => (this.balance = balance));
+
+        this.form.controls.asset.valueChanges
+            .pipe(takeUntil(this.ngUnsubscribe))
+            .subscribe((asset: UserToken | 'ETH') => {
+                this.tokenInput.selectedToken = asset;
+                if (asset === 'ETH') {
+                    this.ethSelected = true;
+                    this.token = undefined;
+                } else {
+                    this.ethSelected = false;
+                    this.token = asset;
+                }
+            });
+        this.form.controls.asset.updateValueAndValidity();
+
+        from(this.web3Factory.detectProvider())
+            .pipe(
+                filter((provider) => !!provider),
+                map((provider: any) => this.web3Factory.create(provider)),
+                tap((web3) => (this.web3 = web3)),
+                switchMap((web3) =>
+                    from<Promise<string[]>>(web3.eth.getAccounts())
+                )
+            )
+            .subscribe((accounts) => {
+                if (accounts.length > 0) {
+                    this.defaultAccount = accounts[0];
+                }
+            });
+    }
+
+    ngOnDestroy() {
+        this.ngUnsubscribe.next();
+        this.ngUnsubscribe.complete();
+    }
+
+    accept() {
+        if (this.defaultAccount) {
+            this.sendTransaction();
+        } else {
+            this.connectWallet();
+        }
+    }
+
+    close() {
+        this.dialogRef.close();
+    }
+
+    private sendTransaction() {
+        const symbol = this.ethSelected ? 'ETH' : this.token.symbol;
+        const formattedAmount = amountToDecimal(
+            this.form.value.amount,
+            this.ethSelected ? 18 : this.token.decimals
+        ).toFixed();
+        let notificationIdentifier: number;
+
+        defer(() => {
+            this.dialogRef.close();
+            notificationIdentifier = this.notificationService.addPendingAction({
+                title: `Sending ${symbol} to account`,
+                description: `${formattedAmount} ${symbol}`,
+                icon: 'deposit',
+                userToken: this.token,
+            });
+            const raidenAddress = this.raidenService.raidenAddress;
+            const transferValue = (
+                this.form.value.amount as BigNumber
```

Comment:
You sure this is a `BigNumber`? Maybe it'd be safer to pass it through the constructor just in case

Reply:
Yes, this value comes from our own implementation of a form input. The `TokenInputComponent` ensures that it is a `BigNumber`. It is just unfortunate that the Angular form API still does not provide any type safety.